python/sdk/test/conftest.py

```diff
@@ -82,7 +82,12 @@ def model(project, mlflow_url, api_client):
 @pytest.fixture
 def version(project, model, mlflow_url, api_client):
     mlflow.set_tracking_uri(mlflow_url)
-    r = mlflow.start_run()
+    run_id = None
+    if not mlflow.get_experiment_by_name("unit_test_experiment"):
+        run_id = mlflow.create_experiment(
+            name="unit_test_experiment"
+        )
+    r = mlflow.start_run(experiment_id=run_id)
```

Comment:
With the removal of pytest parallelism, e2e tests that get run before this test would've already created (and deleted with the changes in this PR) Mlflow experiments, leaving behind a default experiment that does not exist. This causes errors like [these](https://github.com/caraml-dev/merlin/actions/runs/7855799983/job/21437927704#step:15:169) when this test tries to start a new run:
```python3
>           raise MlflowException(
                "Could not find experiment with ID %s" % experiment_id,
                databricks_pb2.RESOURCE_DOES_NOT_EXIST,
            )
E           mlflow.exceptions.MlflowException: Could not find experiment with ID 25
``` 

This happens because `start_run` will "look for valid experiment in the following order: activated using set_experiment, MLFLOW_EXPERIMENT_NAME environment variable, MLFLOW_EXPERIMENT_ID environment variable, or the default experiment as defined by the tracking server" when the `experiment_id` kwarg is not specified (see [this](https://mlflow.org/docs/1.30.0/python_api/mlflow.html#mlflow.start_run) for more details).

To ensure this doesn't happen, we simply force the creation of a new experiment solely for this fixture to create its own Mlflow run.